lib/ansible/modules/cloud/vmware/vmware_datastore_facts.py

```diff
@@ -214,10 +214,12 @@ def get_all_objs(self, content, types, confine_to_datacenter=True):
         if confine_to_datacenter:
             if hasattr(objects, 'items'):
                 # resource pools come back as a dictionary
+                tmpobjs = objects.copy()
```

Comment:
This may not matter if objects only has a small number of entries in it but this is inefficient.  This creates a copy of the dict which includes overhead for the dict's record keeping (hash tables and so forth) when you really only need to have an iterable of the keys and values.  So it's better to keep a copy of objects.items() instead of a copy of the dict itself.  You can do that like this:
``` python
for k, v in tuple(objects.items()):
    parent_dc = get_parent_datacenter(k)
    if parent_dc.name != self.dc_name:
        del objects[k]
```

[*] Note, ```del objects[k]``` should achieve the same thing as ```objects.pop(k, None)``` in this instance.  I like del when simply trying to remove a known key from a dictionary and pop when you want the value back.  (pop is also useful when you don't know if the key is present in the dictionary to begin with as specifying the default value a the second arg means it won't traceback if the key isn't present).

I don't consider this a blocker as I don't know if objects could ever be big enough to be a problem.  I'll go ahead and merge this on Monday if I don't hear back from you.  (And you can always fix it in devel but not backport it if you like the efficiency but don't think that it's important enough to backport).

Reply:
I agree with you, @goneri what do you think ?

Reply:
There is a similar pattern in `vmware_guest` ( https://github.com/ansible/ansible/blob/622a493ae03bd5e5cf517d336fc426e9d12208c7/lib/ansible/modules/cloud/vmware/vmware_guest.py#L788 ).

From my perspective, the `copy()` call just reproduces what was silently done before by Python2. So I don't think it introduces a performance hit.

This being said, if Python3 behaves differently, it's to improve the performance. So we should indeed probably pay attention. I opened an issue to track that: https://github.com/ansible/ansible/issues/60399

Reply:
Actually, my formulation is closer to what happens in Python2 as well.  ```objects.items()``` on Python2 creates a list whereas mine creates a tuple.  (and the code in this PR creates a new dict).

Reply:
It looks like you guys are okay with changing this down the road rather than immediately, tough, so I'll go ahead and merge this for 2.8.4.  Thanks for looking at it.

Reply:
@abadger Thanks.

Reply:
Thanks @abadger, I'm testing the patch that you suggested https://github.com/ansible/ansible/pull/60476.